ts/main.ts

```diff
@@ -78,15 +73,55 @@ function swapView(anime: Anime): void {
     '.detail-rating h2',
   ) as HTMLHeadingElement;
 
+  // Update title
   titleElement.textContent = anime.title;
+
+  // Update synopsis
   synopsisElement.textContent = anime.synopsis;
-  imageElement.innerHTML = `<img src="${anime.images.jpg.image_url}" alt="${anime.title}">`;
+
+  // Update image
+  const img = document.createElement('img');
+  img.src = anime.images.jpg.image_url;
+  img.alt = anime.title;
+
+  // Clear existing children in imageElement and append new image
+  while (imageElement.firstChild) {
+    imageElement.removeChild(imageElement.firstChild);
+  }
+  imageElement.appendChild(img);
+
+  // Update rating
   ratingElement.textContent = `Rating: ${anime.rating ?? 'N/A'}`;
 
-  // Hide the anime container and show the detail view
-  document.querySelector('.anime-container')?.classList.add('hidden');
+  // Show detail view and hide anime container
+  animeContainer.classList.add('hidden');
   detailView.classList.remove('hidden');
 }
 
-// Start the process of fetching and rendering anime data
-fetchAnime();
+// starts the page
+async function init(): Promise<void> {
+  const animeList = await fetchAnime();
+
+  animeList.forEach((anime) => {
+    animeContainer.appendChild(renderAnime(anime));
+  });
+
+  // Event delegation
+  animeContainer.addEventListener('click', (event: Event) => {
+    const target = event.target as HTMLElement;
+    const animeDiv = target.closest('.animes') as HTMLDivElement;
+
+    if (animeDiv) {
+      const animeId = animeDiv.dataset.id;
+      const selectedAnime = animeList.find(
```

Comment:
This would also be better as a guard clause for readability where you run `if (!animeDiv) return` so you immediately exit the function if it is not what you are looking for.